File: report/src/main/java/com/vmlens/report/runelementtype/BarrierKeyTextBuilder.java
package com.vmlens.report.runelementtype;

import com.vmlens.trace.agent.bootstrap.interleave.interleaveactionimpl.barrierkey.BarrierKeyVisitor;
import com.vmlens.trace.agent.bootstrap.interleave.interleaveactionimpl.barrierkey.FutureKey;
import com.vmlens.trace.agent.bootstrap.interleave.interleaveactionimpl.barrierkey.PhaserKey;

public class BarrierKeyTextBuilder implements BarrierKeyVisitor {
    
    private String text = "";
    
    @Override
    public void visit(FutureKey futureKey) {
        text = "Future" ;
    }

    @Override
    public void visit(PhaserKey phaserKey) {
        text = "Phaser";
    }
    
    public String build() {
        return text;
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/BarrierKeyTextBuilder.scala
package com.anarsoft.race.detection.report.element.runelementtype

import com.vmlens.trace.agent.bootstrap.interleave.interleaveaction.barrierkey.{BarrierKeyVisitor, CountDownLatchKey, FutureKey, PhaserKey}


class BarrierKeyTextBuilder extends BarrierKeyVisitor {
  private var text = ""

  override def visit(futureKey: FutureKey): Unit = {
    text = "Future"
  }

  override def visit(phaserKey: PhaserKey): Unit = {
    text = "Phaser"
  }

  override def visit(countDownLatchKey: CountDownLatchKey): Unit = {
    text = "CountDownLatch"
  }

  def build: String = text
}

